### PR TITLE
Return a better error message when the 'coin_id' argument for nft_get_info cannot be decoded into a 32 byte hex string

### PR DESCRIPTION
**Pull request by aqk**
Return a better error message when the 'coin_id' argument for nft_get_info cannot be decoded into a 32 byte hex string.

### Purpose:

Fix error reporting.

### Current Behavior:

Leave an exception in the log when interpreting 'coin_id' as a hex encoded bytes32

### New Behavior:

Return error message `Invalid Coin ID format for 'coin_id': <<contents>>`

### Testing Notes:

See https://github.com/Chia-Network/chia-blockchain/pull/new/ak.fix-14453

Reporter: @esaung

### CODE REVIEW

**Review by paninaro — DISMISSED** [review]

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -2530,7 +2530,10 @@ async def nft_get_info(self, request: Dict) -> EndpointResult:
         if coin_id.startswith(AddressType.NFT.hrp(self.service.config)):
             coin_id = decode_puzzle_hash(coin_id)
         else:
-            coin_id = bytes32.from_hexstr(coin_id)
+            try:
+                coin_id = bytes32.from_hexstr(coin_id)
+            except ValueError:
+                return {"success": False, "error": f"Invalid Coin ID format for 'coin_id' {request['coin_id']}"}
```
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:2536]: I have two basic thoughts around this.  I only expect action here on the first as I think the second could easily turn into a rabbit hole and don't feel the need to force it's exploration.

1. Let's make the information presented a touch more 'formally' to make odd cases of leading spaces etc more apparent.  I'll admit that using `!r` results in a Python representation and that something like `json.dumps()` would be a bit more language agnostic.  Maybe something to consider applying across the whole code base.

```suggestion
                return {"success": False, "error": f"Invalid Coin ID format for 'coin_id': {request['coin_id']!r}"}
```

2. There were two cases listed in https://github.com/Chia-Network/chia-blockchain/issues/14453.  `bad bytes32 initializer b''` and `non-hexadecimal number found in fromhex() arg at position 0`.  This change improves things by always presenting the problematic input, but it does lose the explanation of what the complaint was.

Comment (aqk) [reply chia/rpc/wallet_rpc_api.py:2536]: Good idea for when the values can contain whitespace, yeah.

**Comment by Quexington** [comment]
I think this needs to be rebased, it looks like it brings a lot of main commits with it that are not necessary for the PR.

**Review by Quexington — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Comment by emlowe** [comment]
close and reopen for CI

**Pull request merged**